src/cinder/ImageFileTinyExr.cpp

```diff
@@ -46,119 +49,132 @@ void ImageSourceFileTinyExr::registerSelf()
 }
 
 ImageSourceFileTinyExr::ImageSourceFileTinyExr( DataSourceRef dataSource, ImageSource::Options /*options*/ )
+	: mExrHeader( new EXRHeader, FreeEXRHeader ) // We're using the provided FreeEXRHeader function as a custom deleter
+	, mExrImage( new EXRImage, FreeEXRImage )    // We're using the provided FreeEXRImage function as a custom deleter
 {
-	mExrImage.reset( new EXRImage );
-	const char *error;
-
+	InitEXRHeader( mExrHeader.get() );
 	InitEXRImage( mExrImage.get() );
 
 	int status = 0;
+
+	EXRVersion version;
+	status = ParseEXRVersionFromFile( &version, dataSource->getFilePath().string().c_str() );
+	if( status != TINYEXR_SUCCESS )
+		throw ImageIoExceptionFailedLoadTinyExr( string( "Failed to parse OpenEXR version" ) );
+
+	if( version.multipart || version.non_image )
+		throw ImageIoExceptionFailedLoadTinyExr( string( "Multipart or DeepImage EXR's are not supported yet" ) );
+
 	if( dataSource->isFilePath() ) {
-		status = ParseMultiChannelEXRHeaderFromFile( mExrImage.get(), dataSource->getFilePath().string().c_str(), &error );
-		if( status != 0 )
+		const auto filename = dataSource->getFilePath().string().c_str();
```

Comment:
Looks like this is getting a pointer to the internal storage of a temporary (`string()` is a temporary). I can fix this when I merge.